tutorials/bios-boot-tutorial/README.md

```diff
@@ -7,38 +7,69 @@ The `bios-boot-tutorial.py` script simulates the EOSIO bios boot sequence.
 1. Python 3.x
 2. CMake
 3. git
+4. g++
+5. build-essentials
+6. pip3
+7. openssl
+8. curl
+9. jq
+10. psmisc
+
 
 ``Steps``:
 
 1. Install eosio binaries by following the steps outlined in below tutorial
-[Install eosio binaries](https://github.com/EOSIO/eos#mac-os-x-brew-install)
+[Install eosio binaries](https://github.com/EOSIO/eos/tree/release/2.0.x#mac-os-x-brew-install).
+
+2. Install eosio.cdt version 1.6.3 binaries by following the steps outlined in below tutorial
+[Install eosio.cdt binaries](https://github.com/EOSIO/eosio.cdt/tree/release/1.6.x#binary-releases).
+
+3. Compile `eosio.contracts` version 1.8.x.
+
+```bash
+$ cd ~
+$ git clone https://github.com/EOSIO/eosio.contracts.git eosio.contracts-1.8.x
+$ cd ./eosio.contracts-1.8.x/
+$ git checkout release/1.8.x
+$ ./build.sh
+$ cd ./build/contracts/
+$ pwd
+
+```
+
+4. Make note of the directory where the contracts were compiled. 
```

Comment:
I do not like the idea of storing WASM+ABI in the source repository.

Reply:
I do like the idea of storing WASM+ABI in the source repository and we already do.

Reply:
yeah? where?

Reply:
anyhow, I still do not like it.

Reply:
https://github.com/EOSIO/eos/tree/release/2.0.x/unittests/contracts

Reply:
ok, I see. if we have a precedent which brakes common practice it doesn't mean we have to persist in it. besides the precedent it is in unit tests which is somewhat accepted, however here's the main points against this practice:
1. it is well known in the industry as a general rule that one should not commit binary files in source repo (arguments can be found with a simple google search); with this we do not want to give a bad example.
2. we should not ask our users to trust those binaries we check in and tell them to use them, unless we provide a way to verify them.

Reply:
2. I agree we should provide *optional* instructions/steps for how to generate the wasm files. I just think it would be nice to provide the wasm files so a user doesn't have to go through all the steps needed to generate them unless they want to verify. If I'm just wanting to setup a local node for testing for example.

Reply:
I understand you want to make users life _easier_, but in this case I do not see that _easier_ is also _better_ because there's not much time gain here at the expense of those points I've mentioned, which I see them more important. Also this tutorial it supposed to each every single steps one needs to go through to boot a chain, and this is going to shortcut one of those steps.

Reply:
@heifner I see your points, however, there's a number of issues with this approach.

1. Maintenance: in order to prevent drift, wasm files will have to be compiled with every change to the source of truth which does not exist in this repository (`eosio.contracts`). These binaries will be implicitly tagged by `EOSIO/eos`. This is an anti-pattern and will be prone to drift. 
2. Communication: by storing build artifacts from another repo it can become a source of confusion. It's not unlikely users will begin to use these wasm files which may cause issues especially in the case of drift previously mentioned. 
3. While your suggested pattern is used for unit tests, a subject of a different debate, testing mechanisms generally follow a different set of coding standards and rules than that of source code and documentation. 

In order to resolve this conversation and get this branch merged, could we please settle on the less contentious approach that yields identical results?

Reply:
You only need one approval, it doesn't have to be mine. I completely disagree with your points. 

1. They can drift all they want, the contract is only needed for a few seconds to use for the following steps.

2. Name them something that will avoid confusion, and document what you are doing.

I'm all fine with less contentious approach. So that means you will go with what @arhag and I suggest, yes?

Reply:
@heifner Thanks for your input. 

We do not feel comfortable committing binaries to the repository due to well-documented reasons on the subject. Therefor, we will resolve subsequent changes. 

These changes can be applied in another PR if you all feel comfortable with it.